java/maven.indexer/src/org/netbeans/modules/maven/indexer/NexusRepositoryIndexerImpl.java

```diff
@@ -534,12 +532,22 @@ private void indexLoadedRepo(final RepositoryInfo repo, boolean updateLocal) thr
                         httpwagon.setHttpHeaders(p);
                     }
 
-                    Path tmpStorage = Files.createTempDirectory("index-extraction");
+                    Path tmpStorage = Files.createTempDirectory(getIndexDirectory(), "extractor-");
```

Comment:
note to self: this is extracting directly into cache right now. I am not convinced that this is a good idea since this requires the cleanup on failure/cancel/system.exit to work reliably.

The advantage of this approach is that only the compressed archive will land in tmp - which should fit even into small tmp folders.